src/mixins/filing-template-mixin.ts

```diff
@@ -1562,4 +1578,73 @@ export default class FilingTemplateMixin extends DateMixin {
+    // Documents Info can possibly be undefined, if the co-op was created via paper.
+    const documentsInfo = entitySnapshot.businessDocuments?.documentsInfo
+    if (filingInformation?.rulesFileKey) {
+      // Scenario 1 - From draft, rules are uploaded in the draft.
+      this.setSpecialResolutionRules(
+        {
+          name: filingInformation.rulesFileName,
+          key: filingInformation.rulesFileKey,
+          url: null, // no url for drafts, this is intentional.
+          includedInResolution: false,
+          previouslyInResolution: documentsInfo?.certifiedRules?.includedInResolution,
+          uploaded: filingInformation.rulesUploadedOn
+        })
+    } else if (filingInformation?.rulesInResolution) {
+      // Scenario 2 - From draft, rules are included in the resolution.
+      this.setSpecialResolutionRules(
+        {
+          name: documentsInfo?.certifiedRules?.name,
+          key: documentsInfo?.certifiedRules?.key || null,
+          url: entitySnapshot.businessDocuments.documents?.certifiedRules,
+          includedInResolution: true,
+          previouslyInResolution: documentsInfo?.certifiedRules?.includedInResolution,
+          uploaded: documentsInfo?.certifiedRules?.uploaded
+        })
+    } else {
+      // Scenario 3 + 4 - Not draft, rules exist or rules are on paper.
+      this.setSpecialResolutionRules(
+        {
+          name: documentsInfo?.certifiedRules?.name,
+          key: documentsInfo?.certifiedRules?.key || null,
+          url: entitySnapshot.businessDocuments.documents?.certifiedRules,
+          includedInResolution: false,
+          previouslyInResolution: documentsInfo?.certifiedRules?.includedInResolution,
+          uploaded: documentsInfo?.certifiedRules?.uploaded
+        })
+    }
+
+    if (filingInformation?.memorandumInResolution) {
+      // Scenario 1 - From draft - Memorandum is in the resolution.
+      this.setSpecialResolutionMemorandum(
+        {
+          name: documentsInfo?.certifiedMemorandum?.name,
+          key: documentsInfo?.certifiedMemorandum?.key || null,
+          url: entitySnapshot.businessDocuments.documents?.certifiedMemorandum,
+          includedInResolution: true,
+          previouslyInResolution: documentsInfo?.certifiedMemorandum?.includedInResolution,
+          uploaded: documentsInfo?.certifiedMemorandum?.uploaded
+        })
+    } else {
+      // Scenario 2 + 3 - Not Draft - Memorandum is not in the resolution or are on paper.
+      this.setSpecialResolutionMemorandum(
+        {
+          name: documentsInfo?.certifiedMemorandum?.name,
+          key: documentsInfo?.certifiedMemorandum?.key || null,
+          url: entitySnapshot.businessDocuments.documents?.certifiedMemorandum,
+          includedInResolution: false,
+          previouslyInResolution: documentsInfo?.certifiedMemorandum?.includedInResolution,
+          uploaded: documentsInfo?.certifiedMemorandum?.uploaded
+        })
+    }
+  }
```

Comment:
OK, next time we're in here, I think we should break up this mixin into smaller files :)

Reply:
agreed